config/initializers/kessel.rb

```diff
@@ -5,9 +5,19 @@
 # and validates the configuration when Kessel is enabled.
 
 Rails.application.configure do
-  # Override settings from environment variables
-  if ENV['KESSEL_ENABLED'].present?
-    Settings.kessel.enabled = ActiveModel::Type::Boolean.new.cast(ENV['KESSEL_ENABLED'])
+
+  if defined?(Unleash) && !Unleash.nil?
+    # Check if Unleash is available and enabled
+    if defined?(UNLEASH) && UNLEASH.is_enabled?("kessel_enabled")
```

Comment:
According to the [documentation](https://github.com/Unleash/unleash-ruby-sdk?tab=readme-ov-file#3-sample-usage), I'd say that this could be simplified to

```suggestion   
  if UNLEASH.is_enabled? "kessel_enabled", @unleash_context
```

Also, it seems that the context is not being set, as the 2nd step suggests.